metasrv/src/api/http_service_test.rs

```diff
@@ -29,7 +30,7 @@ use crate::tests::tls_constants::TEST_SERVER_KEY;
 #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
 async fn test_http_service_tls_server() -> Result<()> {
     let mut conf = Config::empty();
-    let addr_str = "127.0.0.1:0";
+    let addr_str = "127.0.0.1:40001".to_string();
```

Comment:
Why aren't we using random port?

Reply:
> Why aren't we using random port?

Let me change to random port and test it again, the test hung when I was coding  for this issue so I changed to a fixed port and forget to change it back.

Reply:
@zhang2014   updated the test, but I still don't understand why the test runs fine in tokio when I bind port 0 and fails without tokio:
failed with `127.0.0.1:0`
```
        srv.start().await.expect("HTTP: admin api error");
        // kick off
        let client = reqwest::Client::builder()
            .add_root_certificate(cert)
            .build()
            .unwrap();
        let resp = client.get(url).send().await;
        assert!(resp.is_ok());
        let resp = resp.unwrap();
        assert!(resp.status().is_success());
        assert_eq!("/v1/health", resp.url().path());
```
works with `127.0.0.1:`
```
tokio::spawn(async move {
        srv.start().await.expect("HTTP: admin api error");
        // kick off
        let client = reqwest::Client::builder()
            .add_root_certificate(cert)
            .build()
            .unwrap();
        let resp = client.get(url).send().await;
        assert!(resp.is_ok());
        let resp = resp.unwrap();
        assert!(resp.status().is_success());
        assert_eq!("/v1/health", resp.url().path());
    });
```